utils/getInvoices/tests/getInvoices.maintenance.test.ts

```diff
@@ -6,108 +6,834 @@ import { ServiceType } from '@utils/constants'
-      invoice: [],
-    }
-    const prevPayment: Partial<IPayment> = null
+  describe('props: { service, company, prevPayment }', () => {
+    it('should NOT load when service = null, company = null, prevPayment = null', () => {
+      const service: Partial<IService> = null
+      const company: Partial<IRealestate> = null
+      const prevPayment: Partial<IPayment> = null
+
+      const invoices = getInvoices({
+        service,
+        company,
+        prevPayment,
+      })
 
-    const invoices = getInvoices({
-      company,
-      service,
-      payment,
-      prevPayment,
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Maintenance })
+      )
     })
+    it('should NOT load when service = null, company = null, prevPayment = { invoice: [Maintenance] }', () => {
+      const service: Partial<IService> = null
+      const company: Partial<IRealestate> = null
+      const prevPayment: Partial<IPayment> = {
+        invoice: [
+          {
+            type: ServiceType.Maintenance,
+            amount: 10,
+            price: 10,
+            sum: 10,
+          },
+        ],
+      }
+
+      const invoices = getInvoices({
+        service,
+        company,
+        prevPayment,
+      })
 
-    expect(invoices).not.toContainEqual(
-      expect.objectContaining({ type: ServiceType.Maintenance })
-    )
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Maintenance })
+      )
+    })
+    it('should NOT load when service = { rentPrice: 10 }, company = null, prevPayment = { invoice: [Maintenance] }', () => {
+      const service: Partial<IService> = {
+        rentPrice: 10,
+      }
+      const company: Partial<IRealestate> = null
+      const prevPayment: Partial<IPayment> = {
+        invoice: [
+          {
+            type: ServiceType.Maintenance,
+            amount: 10,
+            price: 10,
+            sum: 10,
+          },
+        ],
+      }
+
+      const invoices = getInvoices({
+        service,
+        company,
+        prevPayment,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Maintenance })
+      )
+    })
+    it('should NOT load when service = { rentPrice: 10 }, company = null, prevPayment = { invoice: [Electricity] }', () => {
+      const service: Partial<IService> = {
+        rentPrice: 10,
+      }
+      const company: Partial<IRealestate> = null
+      const prevPayment: Partial<IPayment> = {
+        invoice: [
+          {
+            type: ServiceType.Electricity,
+            price: 10,
+            amount: 10,
+            lastAmount: 0,
+            sum: 100,
+          },
+        ],
+      }
+
+      const invoices = getInvoices({
+        service,
+        company,
+        prevPayment,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Maintenance })
+      )
+    })
```

Comment:
Not required in this stage. If we NOT loading invoice even with `prevPayment = { invoice: [Maintenance] }`, so we are completely not loading it with `prevPayment = { invoice: [Electricity] }`

Better to use this kind of tests when we are at stage where we already load `prevPayment = { invoice: [Maintenance] }`

Reply:
> Better to use this kind of tests when we are at stage where we already load prevPayment = { invoice: [Maintenance] }

As you already does at line 444